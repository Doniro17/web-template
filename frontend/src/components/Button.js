import React from 'react';

export default function Input({ children, type, className, onClick, disabled }) {
  return (
    <button
      disabled={disabled}
      data-testid="test-button"
      onClick={onClick}
      // eslint-disable-next-line react/button-has-type
      type={type}
      className={className}
    >
      {children}
    </button>
  );
}
