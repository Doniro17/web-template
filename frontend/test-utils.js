import React from 'react'; // eslint-disable-line no-unused-vars

import { initFontAwesomeLibrary } from './src/utils';

import '@testing-library/jest-dom/extend-expect';

initFontAwesomeLibrary();

// re-export everything
export * from '@testing-library/react';
