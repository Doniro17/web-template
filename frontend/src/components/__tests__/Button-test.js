import { render, fireEvent } from 'test-utils';

import Button from '../Button';

describe('Button', () => {
  it('renders correctly', () => {
    const { container } = render(
      <Button onClick={jest.fn()} type="button" className="button">
        Actual button
      </Button>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('handles onClick event', () => {
    const id = 'test-button';
    const onClickMock = jest.fn();
    const { getByTestId } = render(<Button onClick={onClickMock}>Button</Button>);
    fireEvent.click(getByTestId(id));
    expect(onClickMock).toBeCalledTimes(1);
  });
});
