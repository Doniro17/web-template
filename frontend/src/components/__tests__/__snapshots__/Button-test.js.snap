// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<button
  class="button"
  data-testid="test-button"
  type="button"
>
  Actual button
</button>
`;
